### DISCUSSION TITLE
IKEA Tradfri bulb work only with frontend

### DISCUSSION THREAD

**Discussion by alan888** [Koenkk/zigbee2mqtt#16638] [Devices]
Hello:
I installed a zigbee2mqqt system in a Raspberry pi, basic installation. Then I paired devices: 2 IKEA tradfri bulbs and a Styrbar switch.
In the frontend webpage it works perfectly.
Trying to play with Node-Red it does not worked at all.
I checked the message from the frontend and replicated to see why it does not worked even with the nice green alert saying the message was successfully injected.
Then I moved to mosquitto_sub and mosquitto_pub to check and I found out that messages are received in the correct topic with exactly same characters between mosquitto_pub and the message from the frontend. But it does not work.
I am super lost. If anyone could give me an idea about what could be wrong, it would be great.
Thanks.-

**Comment by HenryWinkelmann** [comment]
Hello alan888 :)
Today I ran into the same issue and spend lots of time.
Unfortunately I can NOT present a way to resolve. 

It is exactly the same with what you described. Every potential reason has been scanned. 
But still it looks like as if zigbee2mqtt does not convert the proper MQTT message to ZIGBEE message. 
I have no other actor available to see if this an IKEA like issue ;)

Let hope there is a nice one to give us a hint how to get rid of this.

**Comment by alan888** [comment]
Hi Henry:
I was able to solve this issue right after I forgot this post.
I ran zigbee2mqtt in debug mode and then I was able to see the real message that was sent to the bulb.
It was of the type:
`msg.payload = { state: "ON" };`
Within the topic like:
`zigbee2mqtt/Light 1/set`
Then the bulb answered to this message with the long string that was shown in mosquitto_sub.
Let me know if this help you.
Cheers.

**Comment by HenryWinkelmann** [comment]
Hi alan888,
this hint saved my live !!! Thank's as much a possible.

Where can I find this topic extention "/set" info, other then in this thread ?
For testing purpose I created the following function node and linked it to a mqtt out node.
`const msgNew = {
    payload: {
        brightness: 254,
        linkquality : 240,
        power_on_behavior: "previous",
        state: "ON",
    },
    topic: "zigbee2mqtt/LED20230408.1/set"
}

return msgNew;`

... and triggered by injection of an kind.
**Without** the topic **/set** in the end it does **not** work ! 
Even if one compares the outputs in MQTT explorer which are 100% identical - but useless

Again thanks - and have a nice (extended) weekend

**Comment by HenryWinkelmann** [comment]
Sorry to post after resolved - but I found the answer to my onw question: 
Where can I find this topic extention "/set" info, other then in this thread ?

**Answer**: simply read the few thousand pages in ZIGBEE2MQTT docs.
https://www.zigbee2mqtt.io/guide/usage/mqtt_topics_and_messages.html
